search/tests/mock_search_engine.py

```diff
@@ -137,6 +137,42 @@ def _process_exclude_dictionary(documents_to_search, exclude_dictionary):
     return documents_to_search
 
 
+def _process_facet_terms(documents, facet_terms):
```

Comment:
So I get what is going on here after reading through the code, but might be helpful to describe the overall flow in either the docstring or a comment block (probably the latter).  Also, if the job of this particular method is to calculate the facet counts, maybe we could name it more specifically.